scalding-serialization/src/main/scala/com/twitter/scalding/serialization/macros/impl/ordered_serialization/providers/StableKnownDirectSubclasses.scala

```diff
@@ -0,0 +1,32 @@
+package com.twitter.scalding.serialization.macros.impl.ordered_serialization.providers
+
+import scala.reflect.macros.whitebox.Context
+
+/**
+ * The `knownDirectSubclasses` method doesn't provide stable ordering
+ * since it returns an unordered `Set` and the `Type` AST nodes don't
+ * override the `hashCode` method, relying on the default identity
+ * `hashCode`.
+ *
+ * This function makes the ordering stable using a list ordered by the
+ * full name of the types.
+ *
+ * Note: because of an implementation detail, immutable Scala sets
+ * preserve the insertion ordering for up to four elements (see `Set1`,
+ * `Set2`, etc). This method maintains the same behavior for backward
+ * compatibility.
```

Comment:
do you think we need this? what are we being compatible with? I'd rather we did the same thing regardless of set size. What if the next version of scala adds more or less special Set implementations?